docs/user-documentation.rst

```diff
@@ -0,0 +1,425 @@
++---------------------------------------+--------+-------------------------------------------+
+| ``SWITCHBOARD_ROOT/add``              | JSON   | NA                                        |
++---------------------------------------+--------+-------------------------------------------+
+| ``SWITCHBOARD_ROOT/update``           | JSON   | NA                                        |
++---------------------------------------+--------+-------------------------------------------+
+| ``SWITCHBOARD_ROOT/status``           | JSON   | NA                                        |
++---------------------------------------+--------+-------------------------------------------+
+| ``SWITCHBOARD_ROOT/delete``           | JSON   | NA                                        |
++---------------------------------------+--------+-------------------------------------------+
+| ``SWITCHBOARD_ROOT/add_condition``    | JSON   | NA                                        |
++---------------------------------------+--------+-------------------------------------------+
+| ``SWITCHBOARD_ROOT/remove_condition`` | JSON   | NA                                        |
++---------------------------------------+--------+-------------------------------------------+
+| ``SWITCHBOARD_ROOT/history``          | JSON   | NA                                        |
++---------------------------------------+--------+-------------------------------------------+
+
+For more details, please look through the example_ code. Once the views are
+defined you should be able to start using switches in your code:
+
+Post-request cleanup
+^^^^^^^^^^^^^^^^^^^^
+
+The last thing to setup is to trigger an event when the request is finished.
+Switchboard needs to cleanup some caching data; if this event is not triggered,
+changes to the switches will not propogate out without server restarts.
+Depending on the framework's architecture, invoking something at the end of a
+request may mean creating some sort of WSGI middleware, or implementing an
+event handler. For example, as WSGI middleware::
+
+    from webob import Request
+    from switchboard.signals import request_finished
+
+    class SwitchboardMiddleware(object):
+
+        def __init__(self, app):
+            self.app = app
+
+        def __call__(self, environ, start_response):
+            req = resp = None
+            try:
+                req = Request(environ)
+                resp = req.get_response(self.app)
+                return resp(environ, start_response)
+            finally:
+                self._end_request(req)
+
+        def _end_request(self, req):
+            if req:
+                # Notify Switchboard that the request is finished
+                request_finished.send(req)
+
+
+Using switches
+==============
+
+Once Switchboard is up and running within the application, it's time to begin
+using switches within the code. By default, Switchboard is set to autocreate
```

Comment:
The autocreate workflow feels a little odd to me because I would expect that one would want to 'prime' the switches ahead of time so that you could set their values. Especially for development. I don't see any documentation here which covers that workflow. Is this not supported?